### DISCUSSION TITLE
How to use GGML_SYCL_F64?

### DISCUSSION THREAD

**Discussion by GermanAizek** [ggml-org/llama.cpp#10807] [Q&A] [answered]
@ggerganov,
Hi again, I am a contributor to your project.
I have server GPU that has an advantage in FP64 tasks, is it possible to compile with such compile flag? Is there a way to rewrite llama.cpp for a larger data type?

**Comment by GermanAizek** [comment]
```
// partial sum for each thread
#ifdef GGML_SYCL_F16
    sycl::half2 tmp = {0.0f, 0.0f}; // two sums for f16 to take advantage of half2 intrinsics
#else
    float tmp = 0.0f;
#endif // GGML_SYCL_F16
```
judging by the code, I can increase all type sizes and append ifdef macros, but I need other contributors to double-check my commits, I may make a mistake.

**Reply by ggerganov** [reply, marked as answer]
It's not that easy. But regardless, we don't plan to support F64.

**Reply by GermanAizek** [reply]
> It's not that easy. But regardless, we don't plan to support F64.

so you won't even accept PR?

**Reply by 0cc4m** [reply]
What's your use case? 64-bit floats are mostly used for scientific simulation work, which is why AMD Instinct cards are common there. But ggml is a machine learning framework and I don't think I have heard of a machine learning advantage to 64-bit floats.

**Reply by qnixsynapse** [reply]
I don't know what's the use case here for supporting fp64? GG**ML** is a **M**achine **L**earning framework. We don't have proper bf16 support yet in SYCL backend.

**Reply by GermanAizek** [reply]
> 64-bit floats are mostly used for scientific simulation work, which is why AMD Instinct cards are common there.

As if you knew what University was using, there really is a large cluster AMD Instinct GPU, and I noticed that many open source projects still do not have an FP64 use case, although this is a really convenient option for some hardware. Main question is, how much do I need to rewrite data types in code? I can do it, I can test it.

**Reply by netrunnereve** [reply]
The Instinct GPUs have faster FP64 compared to consumer GPUs but in the end it's still slower than FP32. I might be wrong here but I think FP64 runs at half the speed of FP32 on AMD professional cards.

If you're running LLMs on the university cluster FP32 will be faster than FP64 and FP16 or a quant will be even faster still.

**Reply by GermanAizek** [reply]
@netrunnereve, I understand that it's easier, but this is more research task.

**Reply by netrunnereve** [reply]
You're welcome to try it for research purposes but good luck finding a model that has FP64 weights in the first place. Also I don't think you'll get much support for this unless you can prove that there are advantages to running LLMs in FP64 (contrary to all the research showing that even FP32 is unnecessary).
